object has no attribute _sa_instance_state

I am trying to upgrade a project from mixer `0.6.3` to the latest version `5.0.6` but I am having some problems with fixtures using the `cycle` shortcut. This is a SQAlchemy project.

The following code fails:

```python
john = mixer.blend(Actor, name='John')
cesar = mixer.blend(Actor, name='Cesar')

mixer.cycle(3).blend(Movie, actor=(x for x in (cesar, cesar, john)))
```

Here is the traceback:

```python
.venv/local/lib/python2.7/site-packages/mixer/main.py:591: in blend
    return type_mixer.blend(**values)
.venv/local/lib/python2.7/site-packages/mixer/main.py:119: in blend
    deferred_values = list(self.fill_fields(target, defaults))
.venv/local/lib/python2.7/site-packages/mixer/main.py:149: in fill_fields
    deferred = self.set_value(target, fname, value)
.venv/local/lib/python2.7/site-packages/mixer/backend/sqlalchemy.py:133: in set_value
    field_value)[1][0])
.venv/local/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2288: in identity_key_from_instance
    self.primary_key_from_instance(instance))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def emit_backref_from_scalar_set_event(state, child, oldchild, initiator):
        if oldchild is child:
            return child
        if oldchild is not None and \
                oldchild is not PASSIVE_NO_RESULT and \
                oldchild is not NEVER_SET:
            # With lazy=None, there's no guarantee that the full collection is
            # present when updating via a backref.
            old_state, old_dict = instance_state(oldchild),\
                instance_dict(oldchild)
            impl = old_state.manager[key].impl
    
            if initiator.impl is not impl or \
                    initiator.op not in (OP_REPLACE, OP_REMOVE):
                impl.pop(old_state,
                         old_dict,
                         state.obj(),
                         parent_impl._append_token,
                         passive=PASSIVE_NO_FETCH)
    
        if child is not None:
>           child_state, child_dict = instance_state(child),\
                instance_dict(child)
E           AttributeError: Mixer (<class 'test.Movie'>): 'generator' object has no attribute '_sa_instance_state'
```

Creating the fixture without the cycle shortcut works file, so I can workaround this problem by using a `for` loop.

I tested this with various Mixer version and it seems that the problem was introduced on version `3.0.0`. It works file with version `2.3.0`.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8331186-object-has-no-attribute-_sa_instance_state?utm_campaign=plugin&utm_content=tracker%2F327725&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F327725&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/mixer/backend/sqlalchemy.py
+++ b/mixer/backend/sqlalchemy.py
@@ -2,6 +2,7 @@
 from __future__ import absolute_import
 
 import datetime
+from types import GeneratorType
 
 import decimal
 from sqlalchemy import func
@@ -59,6 +60,8 @@ class TypeMixer(BaseTypeMixer):
         mixed = []
         for name, deffered in postprocess_values:
             value = deffered.value
+            if isinstance(value, GeneratorType):
+                value = next(value)
             if isinstance(value, t.Mix):
                 mixed.append((name, value))
                 continue

--- a/tests/test_sqlalchemy.py
+++ b/tests/test_sqlalchemy.py
@@ -121,6 +121,16 @@ def test_mixer(session):
     user = mixer.blend(User, username=lambda: 'callable_value')
     assert user.username == 'callable_value'
 
+def test_cycle(session):
+    from mixer.backend.sqlalchemy import Mixer
+
+    mixer = Mixer(session=session, commit=True)
+    profile1 = mixer.blend('tests.test_sqlalchemy.Profile', name='first')
+    profile2 = mixer.blend('tests.test_sqlalchemy.Profile', name='second')
+    users = mixer.cycle(2).blend(User, profile=(p for p in (profile1, profile2)))
+    assert len(users) == 2
+    assert users[0].profile.name == 'first'
+    assert users[1].profile.name == 'second'
 
 def test_select(session):
     from mixer.backend.sqlalchemy import Mixer

